service/migration_manager.cc

```diff
@@ -925,14 +925,15 @@ future<> migration_manager::announce_without_raft(std::vector<mutation> schema)
 
 // Returns a future on the local application of the schema
 future<> migration_manager::announce(std::vector<mutation> schema, group0_guard guard, std::string_view description) {
-    if (is_raft_enabled()) {
-        co_await announce_with_raft(std::move(schema), std::move(guard), std::move(description));
+    if (guard.with_raft()) {
+        return announce_with_raft(std::move(schema), std::move(guard), std::move(description));
     } else {
-        co_await announce_without_raft(std::move(schema));
+        return announce_without_raft(std::move(schema), std::move(guard));
```

Comment:
> If we're using the old ways, we send mutations and return while they're still in flight.

But we only return after these mutations are locally applied - is that right?

If so, then they won't be a problem. Indeed, we leave `synchronize` state only after ensuring that schema is synchronized across all nodes. If there are any in-flight schema mutations, we know they've been applied locally, so we know that at the moment of leaving `synchronize` they've also been applied to all remotes. These in-flight messages may arrive at the remotes after we leave `synchronize`, but they won't have any effect thanks to mutation idempotency.

Reply:
What guarantees that they've been applied to all remotes at the time of leaving synchronize?

Reply:
1. we ensure everyone entered `synchronize`, which means everyone stopped making new schema changes. Any in-flight mutations after this step must have started before this step.
2. we ensure schema is synchronized across all versions. This includes the in-flight mutations (because we apply locally before we send the mutations through PRC).
3. then we leave `synchronzie`.

Nothing guarantees that the in-flight RPCs have reached their destinations by step 3. But the mutations they contain have been applied to every node - in step 2. So when they reach their destinations, they won't have an effect.